contracts/utils/Strings.sol

```diff
@@ -13,26 +13,55 @@ library Strings {
     /**
      * @dev Converts a `uint256` to its ASCII `string` decimal representation.
      */
-    function toString(uint256 value) internal pure returns (string memory) {
-        // Inspired by OraclizeAPI's implementation - MIT licence
-        // https://github.com/oraclize/ethereum-api/blob/b42146b063c7d6ee1358846c198246239e9360e8/oraclizeAPI_0.4.25.sol
+    function toString(uint256 value) internal pure returns (string memory result) {
+        unchecked {
+            if (value < 10) {
+                return string(abi.encodePacked(uint8(value + 48)));
+            }
+            uint256 length = 0;
+            uint256 valueCopy = value;
+            if (valueCopy >= 1000000000000000000000000000000000000000000000000000000000000000) {
+                valueCopy /= 10000000000000000000000000000000000000000000000000000000000000000;
+                length += 64;
+            }
+            if (valueCopy >= 10000000000000000000000000000000) {
+                valueCopy /= 100000000000000000000000000000000;
+                length += 32;
+            }
+            if (valueCopy >= 1000000000000000) {
+                valueCopy /= 10000000000000000;
+                length += 16;
+            }
+            if (valueCopy >= 10000000) {
+                valueCopy /= 100000000;
+                length += 8;
+            }
+            if (valueCopy >= 1000) {
+                valueCopy /= 10000;
+                length += 4;
+            }
+            if (valueCopy >= 10) {
+                valueCopy /= 100;
+                length += 2;
+            }
+            if (valueCopy >= 1) {
+                length += 1;
+            }
```

Comment:
```suggestion
uint256 length = 1;
            uint256 x = value;

            if (x >= 10**64) {
                x /= 10**64;
                length += 64;
            }
            if (x >= 10**32) {
                x /= 10**32;
                length += 32;
            }
            if (x >= 10**16) {
                x /= 10**16;
                length += 16;
            }
            if (x >= 10**8) {
                x /= 10**8;
                length += 8;
            }
            if (x >= 10**4) {
                x /= 10**4;
                length += 4;
            }
            if (x >= 10**2) {
                x /= 10**2;
                length += 2;
            }
            if (x >= 10**1) {
                length += 1;
            }
```

Reply:
Very clever and no gas penalty :+1: Applied.